src/population_loader.rs

```diff
@@ -0,0 +1,140 @@
+use crate::parameters_loader::Parameters;
+use ixa::{
+    context::Context,
+    define_person_property, define_person_property_with_default,
+    error::IxaError,
+    global_properties::ContextGlobalPropertiesExt,
+    people::{ContextPeopleExt, PersonId},
+};
+
+use serde::Deserialize;
+use std::path::PathBuf;
+
+#[derive(Deserialize, Debug)]
+#[allow(non_snake_case)]
+pub struct PeopleRecord<'a> {
+    age: u8,
+    homeId: &'a [u8],
+}
+
+define_person_property!(Age, u8);
+define_person_property!(HomeId, usize);
+define_person_property_with_default!(Alive, bool, true);
+
+define_person_property!(CensusTract, usize);
+
+fn create_person_from_record(
+    context: &mut Context,
+    person_record: &PeopleRecord,
+) -> Result<PersonId, IxaError> {
+    let tract: usize = String::from_utf8(person_record.homeId[..11].to_owned())
+        .expect("Home id should have 11 digits for tract + n digits for home id")
+        .parse()
+        .expect("Could not parse census tract");
+
+    let home_id: usize = String::from_utf8(person_record.homeId.to_owned())
+        .expect("Could not read home id")
+        .parse()
+        .expect("Could not parse home id");
+
+    let person_id = context.add_person((
+        (Age, person_record.age),
+        (HomeId, home_id),
+        (CensusTract, tract),
+    ))?;
+    Ok(person_id)
+}
+
+fn load_synth_population(context: &mut Context, synth_input_file: PathBuf) -> Result<(), IxaError> {
+    let mut reader =
+        csv::Reader::from_path(synth_input_file).expect("Failed to open file. No headers found.");
+    let mut raw_record = csv::ByteRecord::new();
+    let headers = reader.byte_headers().unwrap().clone();
+
+    while reader.read_byte_record(&mut raw_record).unwrap() {
+        let record: PeopleRecord = raw_record
+            .deserialize(Some(&headers))
+            .expect("Failed to parse record.");
+        create_person_from_record(context, &record)?;
+    }
+    Ok(())
+}
+
+pub fn init(context: &mut Context) -> Result<(), IxaError> {
+    let parameters = context
+        .get_global_property_value(Parameters)
+        .unwrap()
+        .clone();
```

Comment:
Is the clone not necessary for keeping context as a mutable borrow when called in `load_synth_population`?

Reply:
No, because you are not passing the the reference to parameters to `load_synth_population()` but rather to `PathBuf::new().  This is most easily seen as follows. The current code is:

```
load_synth_population(context, PathBuf::from(parameters.synth_population_file))?;
```

But this is equivalent to:

```
let tmp = PathBuf::from(parameters.synth_population_file);  // immutable borrow here
load_synth_population(context, tmp)?;   // mutable borrow here.
```

Reply:
I think I'm misunderstanding your original suggestion then because simply removing clone and adding a reference `&parameters.synth_population_file` throws the error that context cannot be called as mutable when called as immutable. Should there be an extra temporary? 

`synth_population_file`  is already known to `parameters` as a `PathBuf`, so I'm not sure if `from` would be the first function called

Reply:
We've had some version skew because the previous version had `Parameters.synth_population_file` as a string and then had the `PathBuf::from()` inside the call to `load_synth_population`, so what I said won't work.

What will work, however, is to clone just the PathBuf:
https://github.com/CDCgov/ixa-epi-isolation/pull/16